### PR TITLE
support torch inference mode for pipelines

### PR DESCRIPTION
**Pull request by nickjyj**
1. In pipeline/predict, we should not calculate gradients. So, we should use torch.inference_mode. 
2. fix some discrepancies between pipeline and docker-compose.yaml

### CODE REVIEW

File: pipeline/pipeline_class.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 
 import logging
 from pipeline_base import PipelineBase as Base
+import torch
```
Comment (trevorditrani) [inline pipeline/pipeline_class.py:14]: If we're importing torch here we should probably install it into the basic docker file

**Review by trevorditrani — APPROVED** [review]
Looks good but can you change the commit message to something more descriptive
